File: projects/proj10/ruby/Kiwi.rb
# Kiwi.rb | Defines the Kiwi class which inherits attributes and methods
#   from the WalkingBird superclass.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Aaron Santucci
# Date: April 28, 2017
####################################################

require './WalkingBird.rb'

class Kiwi < WalkingBird

  def call
    'Daka daka daka'
  end
  
  def className
    self.class.to_s
  end
  
  def print
    puts name + className + ' ' + movement + " by and said " + call
  end
  
end

File: projects/proj10/java/Kiwi.java
/* Kiwi.java provides an Kiwi subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Aaron Santucci
 * Date: May 3, 2017
 ******************************************************/

public class Kiwi extends WalkingBird {

 /* explicit constructor
  * Receive: name, a String 
  * PostCond: myName == name.
  */
	public Kiwi() {
		super();
	}
	
	public Kiwi(String name)
	{
		super(name);
	}



 /* An Kiwi's Call
  * Return: an owl-call ("Whoo-hoo!").
  */
	public String call()
	{
		return "Snork!";
	}

} 

